### PR TITLE
Update file fingerprint

### PR DESCRIPTION
**Pull request by JonoYang**
This PR updates the file fingerprint checksum to also include the number of elements that went into the hash. New tests have been added to profile BitAverageHaloHash behavior

### CODE REVIEW

File: tests/test_halohash.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    field_names, expected = load_csv(expected_file)
+    key1 = field_names[0]
+    # then check results line by line for more compact results
+    for exp, res in zip(expected, results):
+        assert exp[key1] == res[key1]
+        expected_mean_hamming_distance = exp['mean hamming distance']
+        exp_min = expected_mean_hamming_distance - exp['standard deviation']
+        exp_max = expected_mean_hamming_distance + exp['standard deviation']
+        assert exp_min <= res['mean hamming distance']
```
Comment (JonoYang) [inline tests/test_halohash.py:69]: @pombredanne I'm having trouble getting tests to pass, where the mean hamming distance value of our test run is larger or smaller than the expected maximum or minimum hamming distance value.

```
            ...
            assert exp_min <= res['mean hamming distance']
>           assert res['mean hamming distance'] <= exp_max
E           assert 9.4 <= 9.299999999999999
```

Do you have any suggestions on how to best compare results?

**Review by JonoYang — COMMENTED** [review]
@pombredanne I have a question about testing the average hamming distance values

File: tests/test_halohash.py (hunk 1 above)
Comment (pombredanne) [reply tests/test_halohash.py:69]: I would try to bump the acceptable range to be +/-20%

Comment (JonoYang) [reply tests/test_halohash.py:69]: @pombredanne The same issue is happening:
```
            expected_mean_hamming_distance = exp['mean hamming distance']
            exp_min = expected_mean_hamming_distance * 0.8
            exp_max = expected_mean_hamming_distance * 1.2
            assert exp_min <= res['mean hamming distance']
>           assert res['mean hamming distance'] <= exp_max
E           assert 6.2 <= 5.88
```

Comment (JonoYang) [reply tests/test_halohash.py]: Increasing number of tests per word removed/replaced gives us a better standard deviation.

**Pull request merged**